### ISSUE TITLE
ttnn.matmul with transpose_a=True produces incorrect results

### ISSUE THREAD

**Issue by moritztng** [tenstorrent/tt-metal#35247]
### Component / Area

ops

### Issue Type (optional)

Bad Outputs

### Observed

When using ttnn.matmul with transpose_a=True, the result is incorrect. The last row of the output tensor contains zeros instead of the correct values.

Example with input tensors a and b both of shape [2, 3]:
- Correct result (using ttnn.permute to transpose a first): [[17., 22., 27.], [22., 29., 36.], [27., 36., 45.]]
- Incorrect result (using transpose_a=True): [[17., 22., 27.], [22., 29., 36.], [0., 0., 0.]]

Note: transpose_b=True works correctly, so the issue is specific to transpose_a=True.

### Expected

Using transpose_a=True should produce the same result as explicitly transposing the first operand with ttnn.permute before the matmul. The output should be [[17., 22., 27.], [22., 29., 36.], [27., 36., 45.]] with no zeros.

### 1. Steps (exact commands)

```
#!/usr/bin/env python3
import torch
import ttnn

device = ttnn.open_device(device_id=0)

a = ttnn.from_torch(torch.tensor([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]]), dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
b = ttnn.from_torch(torch.tensor([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]]), dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)

# Correct: no transpose
a_perm = ttnn.permute(a, (1, 0))
result_correct = ttnn.to_torch(ttnn.matmul(a_perm, b))
print("Correct (no transpose):\n", result_correct)

# Correct: transpose_b works
result_transpose_b = ttnn.to_torch(ttnn.matmul(a, b, transpose_b=True))
print("Correct (transpose_b):\n", result_transpose_b)

# Incorrect: transpose_a
result_transpose_a = ttnn.to_torch(ttnn.matmul(a, b, transpose_a=True))
print("Incorrect (transpose_a):\n", result_transpose_a)

ttnn.close_device(device)
```

Output:
```
Correct (no transpose):
 tensor([[17., 22., 27.],
        [22., 29., 36.],
        [27., 36., 45.]], dtype=torch.bfloat16)
Correct (transpose_b):
 tensor([[14., 32.],
        [32., 77.]], dtype=torch.bfloat16)
Incorrect (transpose_a):
 tensor([[17., 22., 27.],
        [22., 29., 36.],
        [ 0.,  0.,  0.]], dtype=torch.bfloat16)
```

### 2. Input data / link or description

Tensor a and b in the script.

### 3. Frequency

Always

### 1. Software Versions

Debian 12
TT-Metal commit: 577851495531407f217453c0fd541c253464e9a7
TT-NN version: 0.62.0rc36.dev3035

### 2. Hardware Details

Blackhole p150

### Is this a regression?

Unknown

### Priority

P1

**Comment by jasondavies** [comment]
Could be a regression introduced by #32295 ?  I haven't checked -- try running the test on a prior commit?

**Comment by bbradelTT** [comment]
If there's something else that's failing, please post a pytest showing the incorrect behaviour.

**Comment by moritztng** [comment]
Reopening with a new repro case: transpose_a=True still appears incorrect when the second input tensor (matrix B / right-hand-side input) uses low-precision dtypes.

- transpose_a=True still diverges from explicit transpose when matrix B uses low-precision dtypes.
- I reduced matrix B to a binary mask (0/1) so quantization ambiguity is removed; issue still reproduces.
- Results:
    - matrix B = bfloat16: match (max_abs_diff=0)
    - matrix B = bfloat8_b: mismatch (max_abs_diff=7.6e18)
    - matrix B = bfloat4_b: mismatch (max_abs_diff=inf)

```python
#!/usr/bin/env python3
"""Minimal repro: transpose_a mismatch with low-precision RHS.

Expected:
  matmul(permute(a), b) == matmul(a, b, transpose_a=True)

Observed:
  Matches for RHS bfloat16, but diverges for RHS bfloat8_b / bfloat4_b.
"""

import torch
import ttnn


def run_case(device, rhs_dtype):
    torch.manual_seed(0)
    ck = ttnn.types.BlackholeComputeKernelConfig(
        math_fidelity=ttnn.MathFidelity.HiFi4,
        math_approx_mode=False,
        fp32_dest_acc_en=True,
        packer_l1_acc=True,
    )

    # Use M_per_batch=1 so transpose_a path is accepted by runtime.
    # Shapes are intentionally tiny and easy to reason about.
    # a0: [B, K, 1], b: [B, N]
    B, K, N = 32, 16, 128
    a0_pt = 0.01 * torch.randn((B, K, 1), dtype=torch.bfloat16)
    # Use a binary mask-like matrix B so values are exactly representable
    # across bfloat16 / bfloat8_b / bfloat4_b.
    b_pt = torch.randint(0, 2, (B, N), dtype=torch.int32).to(torch.bfloat16)

    a0 = ttnn.from_torch(a0_pt, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
    b = ttnn.from_torch(b_pt, dtype=rhs_dtype, layout=ttnn.TILE_LAYOUT, device=device)

    # Reference: explicit transpose in layout.
    out_ref = ttnn.to_torch(
        ttnn.matmul(
            ttnn.permute(a0, (2, 1, 0)),  # [1, K, B]
            b,                            # [B, N]
            compute_kernel_config=ck,
            core_grid=ttnn.CoreGrid(y=10, x=13),
        )
    )

    # Candidate: inline transpose_a.
    out_inline = ttnn.to_torch(
        ttnn.matmul(
            ttnn.permute(a0, (2, 0, 1)),  # [1, B, K]
            b,                            # [B, N]
            transpose_a=True,
            compute_kernel_config=ck,
            core_grid=ttnn.CoreGrid(y=10, x=13),
        )
    )

    diff = (out_ref.float() - out_inline.float()).abs()
    print(f"RHS={rhs_dtype}")
    print(f"  max_abs_diff={diff.max().item():.6g}")
    print(f"  mean_abs_diff={diff.mean().item():.6g}")
    print(f"  match={bool(diff.max().item() <= 1e-3)}")


def main():
    device = ttnn.open_device(device_id=0)
    try:
        for dt in (ttnn.bfloat16, ttnn.bfloat8_b, ttnn.bfloat4_b):
            run_case(device, dt)
    finally:
        ttnn.close_device(device)


if __name__ == "__main__":
    main()
```
Output:
```bash
RHS=DataType.BFLOAT16
  max_abs_diff=0
  mean_abs_diff=0
  match=True
RHS=DataType.BFLOAT8_B
  max_abs_diff=7.60208e+18
  mean_abs_diff=1.80304e+18
  match=False
RHS=DataType.BFLOAT4_B
  max_abs_diff=inf
  mean_abs_diff=inf
  match=False
```
